packages/browser-client/src/peerActions.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  ENR,
+  fromHexString,
+  HistoryNetworkContentKeyUnionType,
+  HistoryNetworkContentTypes,
+  HistoryProtocol,
+  reassembleBlock,
+} from 'portalnetwork'
+import { PeerContextType, PeerDispatch, PeerState, PeerStateChange } from './peerReducer'
+
+export class PeerActions {
+  state: PeerState
+  dispatch: PeerDispatch
+  historyProtocol: HistoryProtocol
+  constructor(peerContext: PeerContextType, protocol: HistoryProtocol) {
+    this.state = peerContext.peerState
+    this.dispatch = peerContext.peerDispatch
+    this.historyProtocol = protocol
+  }
+
+  addToOffer = (type: string): void => {
```

Comment:
If we remove the "Offer Block" buttons, we can simplify this whole method to just accept a `HistoryNetworkContentTypes` "type" parameter and then put that value in the `selector` field of the `HistoryNetworkContentKeyUnionType...` serializer and remove the whole switch statement.